inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/detection_output.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "single_layer_tests/detection_output.hpp"
+
+using namespace LayerTestsDefinitions;
+
+namespace {
+
+const size_t numClasses = 11;
+const std::vector<int> topK = {75};
+const std::vector<std::vector<int>> keepTopK = { {50}, {100} };
+const std::vector<std::string> codeType = {"caffe.PriorBoxParameter.CORNER", "caffe.PriorBoxParameter.CENTER_SIZE"};
+const std::vector<bool> clipAfterNms = {true, false};
+const std::vector<bool> clipBeforeNms = {true, false};
+const std::vector<bool> decreaseLabelId = {true, false};
+const std::vector<size_t> numberBatch = {1, 2};
+
+const auto commonAttributes = ::testing::Combine(
+        ::testing::Values(numClasses),
+        ::testing::ValuesIn(topK),
+        ::testing::ValuesIn(keepTopK),
+        ::testing::ValuesIn(codeType),
+        ::testing::ValuesIn(clipAfterNms),
+        ::testing::ValuesIn(clipBeforeNms),
+        ::testing::ValuesIn(decreaseLabelId)
+);
+
+/* =============== 3 inputs cases =============== */
+
+const std::vector<ParamsWhichSizeDepends> specificParams3In = {
+    ParamsWhichSizeDepends{true, true, true, 1, 1, {1, 60}, {1, 165}, {1, 1, 60}, {}, {}},
+    ParamsWhichSizeDepends{true, false, true, 1, 1, {1, 660}, {1, 165}, {1, 1, 60}, {}, {}},
+    ParamsWhichSizeDepends{false, true, true, 1, 1, {1, 60}, {1, 165}, {1, 2, 60}, {}, {}},
+    ParamsWhichSizeDepends{false, false, true, 1, 1, {1, 660}, {1, 165}, {1, 2, 60}, {}, {}},
+
+    ParamsWhichSizeDepends{true, true, false, 10, 10, {1, 60}, {1, 165}, {1, 1, 75}, {}, {}},
+    ParamsWhichSizeDepends{true, false, false, 10, 10, {1, 660}, {1, 165}, {1, 1, 75}, {}, {}},
+    ParamsWhichSizeDepends{false, true, false, 10, 10, {1, 60}, {1, 165}, {1, 2, 75}, {}, {}},
+    ParamsWhichSizeDepends{false, false, false, 10, 10, {1, 660}, {1, 165}, {1, 2, 75}, {}, {}}
+};
+
+const auto params3Inputs = ::testing::Combine(
+        commonAttributes,
+        ::testing::ValuesIn(specificParams3In),
+        ::testing::ValuesIn(numberBatch),
+        ::testing::Values(CommonTestUtils::DEVICE_CPU)
+);
+
+INSTANTIATE_TEST_CASE_P(smoke_DetectionOutput3In, DetectionOutputLayerTest, params3Inputs, DetectionOutputLayerTest::getTestCaseName);
```

Comment:
Could you please try to add instance for GPU as well?

Reply:
for GPU test instance call [detection_output_cpu](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp#L46)
There are any problem
In [NMS](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp#L197) should be check on negative intersect width and height value [Intel Caffe](https://github.com/intel/caffe/blob/3f494b442ee3f9d17a07b09ecbd5fa2bbda00836/src/caffe/util/bbox_util.cpp#L757)
As far as I understand in case then ```normalize == false``` should not compute offset for ```variance_encoded``` considering ```prior_info_size``` and ```prior_coordinates_offset``` [OpenCL kernel](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/include/detection_output_common.cl#L100)
Also should not mix sort type to escape situation when results with the same ```confidence``` swap in ```detection_output_cpu``` and not swap in IntelCaffe or vice versa. In [IntelCaffe](https://github.com/intel/caffe/blob/master/src/caffe/layers/detection_output_layer.cpp#L314) used ```std::sort```.